color_refinement.py

```diff
@@ -78,6 +80,19 @@ def color_refine(coloring: "Coloring") -> "Coloring":
     return coloring
 
 
+def generate_neighbour_count_with_color(graph, current_color):
+    counter = {}
+    for v in graph.vertices:
+        count = 0
+        for x in v.neighbours:
+            if x.colornum is current_color:
+                count += 1
+        if v.colornum not in counter.keys():
+            counter[v.colornum] = {}
+        counter[v.colornum].update({v: count})
```

Comment:
Is `counter[v.colornum][v] = count` niet hetzelfde? En als dat zo is, is dat misschien sneller dan een `dict` instantiëren met `{v: count}` en die vervolges in een andere uitpakken en opslaan.